### ISSUE TITLE
Option to override cookie domain

### ISSUE THREAD

**Issue by Soviut** [auth0/auth0-vue#90]
### Describe the problem you'd like to have solved

We have multiple apps, each on their own subdomain. We want users to be able to sign in on any of these subdomains and remain signed in when visiting the other subdomains.

Currently, a cookie is created with a domain that includes the subdomain. This is too specific;  the cookie will only exist on that subdomain and none of the others. In order for the cookie to span multiple subdomains it must be written as `domain=.site.com`.

### Describe the ideal solution

Ideally, I would be able to set a `cookieDomain` as part of the `createAuth0` client options. This could be passed as [`ClientStorageOption`](https://github.com/auth0/auth0-spa-js/blob/50159871db2dd27329ef7df6d87ca31a85d048d3/src/storage.ts#L3) to the [`CookieStorage.save()`](https://github.com/auth0/auth0-spa-js/blob/50159871db2dd27329ef7df6d87ca31a85d048d3/src/storage.ts#L30) method.  There the domain can be added to the `cookieAttributes` when the cookie is created.

This would allow a top level domain cookie to be created with all the correct attributes, including the correct expiry date, that is accessible across multiple subdomains.

## Alternatives and current work-arounds

Currently, I use a hacky work around that involves performing the `handleRedirectCallback()` manually on my callback page. I then create a cookie with the same name but the desired top level domain.

```
auth0.<clientid>.is.authenticated=true;domain=.site.com;path=/;expires=${date};
```

This leaves me with 3 cookies; the real auth0 cookie, the legacy auth0 cookie and my top level domain cookie.

When I visit another subdomain where I haven't explicitly logged in, the top level domain cookie is present and the auth0 SDK recognizes it, assumes its logged in, and performs a `checkSession`, creating the auth0 and auth0 legacy cookies in the process.

On log out I have to remember to remove the top level cookie before calling `auth0.logout()`.

Having to manually keep track of extra cookies when logging in and out is not ideal. I have to manually set the expiry date based on what's set in the Auth0 tenant configurations.

**Comment by Soviut** [comment]
I'm going to attempt a PR for this to hopefully expedite the process.

**Comment by Soviut** [comment]
Here's the PR. https://github.com/auth0/auth0-spa-js/pull/885

I realized that all the CookieStorage stuff is actually  on the https://github.com/auth0/auth0-spa-js repo.

**Comment by frederikprijck** [comment]
First of all, Thank you so much for all these feedback.
It's both great to see our beta version being used, but also getting very useful feedback to address from our end.

I will leave the conversation for whether or not we should make the cookie domain configurable for the PR in the SPA-JS repo.

But I do want to highlight that you can achieve what you want by calling `getAccessTokenSilently()` when your application starts.

When our SDK is initialized, the first thing it does is call `checkSession`: https://github.com/auth0/auth0-spa-js/blob/master/src/Auth0Client.ts#L751-L772

In short, all this does is check for the cookie, and if it exists call `getTokenSilently()`.
Nothing stops you from calling `getTokenSIlently()` (or `getAccessTokenSilently()` in our Vue, React and Angular SDK) yourself, without guarding it with the cookie check.
Doing so would allow for SSO on multiple applications.

**Comment by Soviut** [comment]
I tried this across two subdomains and, as I recall, `getAccessTokenSilently()` threw a "not logged in" error.

**Comment by frederikprijck** [comment]
If that is the case, any configurable cookie domain wont change anything...

**Comment by Soviut** [comment]
I see. I can definitely test that. However, having a top-level cookie dramatically simplifies things.

It lets me use all the same mechanisms the SDK provides with only a config change instead of a code change.
It saves at least one round trip to Auth0.
It's easier to debug.

**Comment by frederikprijck** [comment]
Let's keep that conversation to the SPA-JS repository. Our Vue SDK is a wrapper for SPA-JS and we follow what's implemented in there.

Just to set expectations, this isn't something we will be looking into before doing a GA release for Auth0-Vue, because the functionality you describe has been like this for years in SPA-JS and shouldn't block our Vue SDK release. For the time being, you will need to use what I mentioned above.

That said, let's see where the conversation in SPA-JS goes and what the team thinks of the idea.

**Comment by frederikprijck** [comment]
I am going to close this issue as we believe it does not need direct action here. However, I had a conversation with the team and it sounds like it could be a good idea to consider the configurable cookiename in SPA-JS.
